### DISCUSSION TITLE
Getting Replex working consistently.

### DISCUSSION THREAD

**Discussion by ericfri** [lostb1t/replex#83] [Q&A] [answered]
First, thank you. Replex is fantastic and I love having the hero style rows. But I'm having a real issue getting it to work consistently. I will go directly to https://replex.mydomain.com and it will work for a while, but then it stops and doesn't come back. Sometimes it will if I log out and back in, but usually only after a restart of the Docker container. 
Do you have any idea why it wouldn't be working consistently?

I am using a reverse proxy with NginxProxyManager, disabled remote access, changed my Custom server access URL to be only https://replex.mydomain.com, and yet, sometimes it's there and other times it's not.

**Comment by lostb1t** [comment]
What exactly do you mean with it.isn't there? The hero rows don't show anymore? Or mixed rows don't work anymore?

You can enable the env setting RUST_LOG="info,replex=trace" to see if Replex still handles incoming requests when you encounter this.

Also check the log for errors

**Comment by ericfri** [comment]
Yes, the hero rows work for a while, then don't. I've got a "trending" collection in both TV and movies, and they both don't show as hero and are no longer combined. Things go back to as if Replex was disabled.
I've added the logging as suggested, and there is a lot. Is there anything in particular I should be looking for?

I just restarted Replex and opened an incognito window. Loading https://replex.mydomain.com didn't show the hero rows but going to https://app.plex.tv/desktop did show them. I will copy both to try and compare.

**Reply by lostb1t** [reply]
The merging strategy is a default so if that doesn't work it seems your requests aren't routed through Replex. I suggest taking a closer look at your proxy.

**Reply by ericfri** [reply]
It seems to be mostly (all?) a local issue and I'm not sure what's happening.
I turn off remote access, uPNP, disable the port forward, turn off both replex and my proxy, and yet I can still browse my content locally via app.plex.tv. If I use my phone and use my LTE, then nothing works until I turn the proxy and replex back on, at which point the hero rows show up on my phone. I just tried tethering my computer to my phone and the hero rows didn't show up.

I'm really unsure of what's happening and why it's so intermittent.

**Reply by jl94x4** [reply]
Sounds like your proxy is not open to traffic outside your server/local network?

**Comment by lostb1t** [comment]
Disable gdm in the network: "Enable local network discovery (GDM)" in plex

Also if running docker for plex don't use the host network option. As it will bind to your server ip. Just open the 32400 port with docker

Make sure to clear a clients app afterwards (if possible)

If that doesn't work you can use this url

https://clients.plex.tv/api/v2/resources?includeIPv6=1&includeRelay=1&X-Plex-Language=en-NL&X-Plex-Token=YOURTOKEN&X-Plex-Client-Identifier=1234

(Chance token to your token)

To see I plex reports the correct server address to your clients

**Comment by ericfri** [comment]
I already had GDM disabled.
Docker is using a bridge network, and that is working correctly.
Looking over the URL, I see two hosts, which seems like it could be the issue.

```
<connections>
<connection protocol="http" address="10.3.37.20" port="32400" uri="http://10.3.37.20:32400" local="1" relay="0" IPv6="0"/>
<connection protocol="https" address="replex.mydomain.com" port="443" uri="https://replex.mydomain.com" local="0" relay="0" IPv6="0"/>
</connections>
```

I'm not sure how to disable the local option.

**Comment by lostb1t** [comment]
That local version is your problem. Plex should not be reachable outside of your proxy entrypoint

**Reply by ericfri** [reply]
That makes sense, but I can't figure out how to disable it.

**Reply by lostb1t** [reply]
Either put nginx manager and plex/Replex on the same virtual network that's not public accesible or try to bind plex to local instead of 0.0.0.0 (default)

"-p 127.0.0.1:32400:32400"

**Comment by Zorro92** [comment]
Are you using docker compose or a docker command?

**Comment by ericfri** [comment, marked as answer]
I'm using Unraid and a template from the Community Apps store and I think I've solved it.

I first created another Docker network `docker network create <network name>` and moved Plex to this network.
Then in my Replex settings, I added the command `&& docker network connect <network> <container>` to the `Post Arguments` under Advanced settings and changed the `Plex Host` to `http://Plex-Media-Server:32400` where `Plex-Media-Server` is the name of the Plex container. By adding the post command, Replex sits on both networks.

This has put Plex on a different network, a 172.18.0.0/16 network and connecting Replex to the same network allows the containers to talk via Hostname (container name).

I added my Replex IP to the `Custom server access URLs`, so I have both my public and private options.
`https://replex.mydomain.com, http://10.3.37.38:80` which changes the connections published to the following.
```
<connections>
<connection protocol="https" address="replex.mydomain.com" port="443" uri="https://replex.mydomain.com" local="0" relay="0" IPv6="0"/>
<connection protocol="http" address="10.3.37.38" port="80" uri="http://10.3.37.38:80" local="1" relay="0" IPv6="0"/>
<connection protocol="http" address="172.18.0.3" port="32400" uri="http://172.18.0.3:32400" local="1" relay="0" IPv6="0"/>
</connections>
```
`10.3.37.38` is the IP of my Replex container and `172.18.0.3` is the IP of my Plex container. Since none of my clients will have a 172 address, they won't be able to talk this way.

I got most of this from https://www.reddit.com/r/unRAID/comments/q2ojh5/using_multiple_docker_networks_in_unraid/

**Comment by ericfri** [comment]
I seem to have a new issue.

On Android specifically, the artwork is no longer showing for any items except the Hero row. Looking through the debug logs, I'm seeing a 404 error for this URL.
`http://replex.mydomain.com/photo/:/transcode?width=330&url=http%3A%2F%2F127.0.0.1%3A80%2Flibrary%2Fmetadata%2F120552%2Fthumb%2F1694522555&height=495&X-Plex-Token=TOKEN&X-Plex-Language=en-ca`
but if I change the URL from 127.0.0.1 to my Plex host, `Plex-Media-Server`, the image loads fine. Is this a bug or is there a way to update that URL?

**Reply by ericfri** [reply]
I had tried adding my local replex IP in my server URLs. Once I removed that things started working again.
